lib/routes/charge/sale_view.dart

```diff
@@ -0,0 +1,212 @@
+        if (sale.saleLines.length == 0) {
+          Navigator.of(context).pop();
+        }
+      });
+    }
+    super.didChangeDependencies();
+  }
+
+  @override 
+  void dispose() {
+    _currentSaleSubscrription.cancel();
+    super.dispose();
+  }
+  
+  @override
+  Widget build(BuildContext context) {
+    PosCatalogBloc posCatalogBloc =
+        AppBlocsProvider.of<PosCatalogBloc>(context);
+    AccountBloc accountBloc = AppBlocsProvider.of<AccountBloc>(context);
+
+    return StreamBuilder<Sale>(
+        stream: posCatalogBloc.currentSaleStream,
+        builder: (context, saleSnapshot) {
+          var currentSale = saleSnapshot.data;
+          return StreamBuilder<AccountModel>(
+              stream: accountBloc.accountStream,
+              builder: (context, accSnapshot) {
+                var accModel = accSnapshot.data;
+                if (accModel == null) {
+                  return Loader();
+                }
+                return Scaffold(
+                  appBar: AppBar(
+                    iconTheme: Theme.of(context).appBarTheme.iconTheme,
+                    textTheme: Theme.of(context).appBarTheme.textTheme,
+                    backgroundColor: Theme.of(context).canvasColor,
+                    leading: backBtn.BackButton(iconData: Icons.close),
+                    title: _TotalSaleCharge(
+                      accountModel: accModel,
+                      currentSale: currentSale,
+                      useFiat: widget.useFiat,
+                    ),
+                    actions: <Widget>[
+                      IconButton(
+                        icon: Icon(
+                          Icons.note_add,
+                          color: Theme.of(context).iconTheme.color,
+                        ),
+                        onPressed: () {},
+                      )
+                    ],
+                    elevation: 0.0,
+                  ),
+                  body: SaleLinesList(
+                      accountModel: accModel, currentSale: currentSale),
+                  bottomNavigationBar: SingleButtonBottomBar(
+                    text: "Clear Sale",
+                    onPressed: () {
+                      posCatalogBloc.actionsSink
+                          .add(SetCurrentSale(Sale(saleLines: [])));
+                    },
+                  ),
+                );
+              });
+        });
+  }
+}
+
+class _TotalSaleCharge extends StatelessWidget {
+  final AccountModel accountModel;
+  final Sale currentSale;
+  final bool useFiat;
+
+  const _TotalSaleCharge(
+      {Key key, this.accountModel, this.currentSale, this.useFiat})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var totalSats = currentSale.totalChargeSat;
+    String formattedCharge;
+    if (useFiat) {
+      formattedCharge = accountModel.fiatCurrency.format(totalSats);
+    } else {
+      formattedCharge = accountModel.currency.format(totalSats);
+    }
+    return Text(
```

Comment:
Return empty string for 0 total charge to prevent unwanted behaviour* on deleting last item.

*: Displaying "< €0.00" right before popping the widget.

Reply:
Good feedback. I changed to format correctly.